kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,173 @@
+            plots_dict = {}
+            for colour in ["blue", "green", "red"]:
+                plots_dict[colour] = plt.figure()
+                plt.plot([1, 2, 3], [4, 5, 6], color=colour)
+                plt.close()
+
+            dict_plot_writer = MatplotlibWriterS3(
+                bucket="my-super-great-bucket",
+                s3_put_object_args={"ServerSideEncryption": "AES256"},
+                filepath="matplotlib_dict",
+            )
+            dict_plot_writer.save(plots_dict)
+
+            # Saving list of plots
+            plots_list = []
+            for index in range(5):
+                plots_list.append(plt.figure())
+                plt.plot([1,2,3],[4,5,6], color=colour)
+            list_plot_writer = MatplotlibWriterS3(
+                bucket="my-super-great-bucket", filepath="matplotlib_list"
+            )
+            list_plot_writer.save(plots_list)
+
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        bucket: str,
+        filepath: str,
+        boto_session_args: Optional[Dict[str, Any]] = None,
+        s3_client_args: Optional[Dict[str, Any]] = None,
+        s3_put_object_args: Optional[Dict[str, Any]] = None,
+        credentials: Optional[Dict[str, Any]] = None,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            bucket: Name of the bucket without "s3://" prefix
+            filepath: Path to a matplot object file.
+            boto_session_args: See
+                https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html
+            s3_client_args: See
+                https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#client
+            s3_put_object_args: See
+                https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.put_object
+            credentials: A dictionary of s3 access and secret keys.
+                Must contain ``aws_access_key_id`` and ``aws_secret_access_key``.
+                Updates ``s3_client_args`` if provided.
+            load_args: Currently ignored as loading is not supported.
+            save_args: Save args passed to `plt.savefig`. See
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+
+        self._boto_session_args = boto_session_args or {}
+        self._s3_client_args = s3_client_args or {}
+        self._credentials = copy.deepcopy(credentials) or {}
+        self._s3_put_object_args = s3_put_object_args or {}
+
+        if self._credentials:
+            self._s3_client_args["aws_access_key_id"] = self._credentials[
+                "aws_access_key_id"
+            ]
+            self._s3_client_args["aws_secret_access_key"] = self._credentials[
+                "aws_secret_access_key"
+            ]
```

Comment:
Since this is the only place where the credentials are used, we don't have to store them on `self`, but use `_credentials` like suggested on line above.

Reply:
Makes sense!